In [7]:
import pandas as pd
import numpy as np

In [8]:
#from statsmodels.stats.weightstats import ztest as ztest
#ztest(df['Cliente A'], df['Cliente B'])

# Prueba T


## Distribución t-student

In [10]:
!pip install pingouin
import pingouin as pg

# Descarga de datos
url = ('https://raw.githubusercontent.com/JoaquinAmatRodrigo/' +
       'Estadistica-machine-learning-python/master/data/births.csv')

datos = pd.read_csv(url, sep=',')
datos.head()

f_age  m_age  weeks  premature  visits  gained  weight sex_baby      smoke
0   31.0     30     39  full term    13.0     1.0    6.88     male     smoker
1   34.0     36     39  full term     5.0    35.0    7.69     male  nonsmoker
2   36.0     35     40  full term    12.0    29.0    8.88     male  nonsmoker
3   41.0     40     40  full term    13.0    30.0    9.00   female  nonsmoker
4   42.0     37     40  full term     NaN    10.0    7.94     male  nonsmoker

In [ ]:
# Filtrar la base de datos entre madres fumadoras y no fumadoras
peso_smokers = datos.loc[datos.smoke == 'smoker', 'weight']
peso_nonsmokers = datos.loc[datos.smoke == 'nonsmoker', 'weight']
peso_smokers

In [14]:
# aplicando el t-test
pg.ttest(x = peso_smokers, y = peso_nonsmokers)

T        dof alternative     p-val          CI95%   cohen-d  \
T-test -1.496711  89.277413   two-sided  0.137995  [-0.93, 0.13]  0.268758   

         BF10     power  
T-test  0.512  0.338075

Como el $p-valor = 0.13>0.05$

$T = -1.4$ esta en la region de aceptacion

Es decir $μ_{1} = μ_{2}$

Por lo tanto, no hay diferencia entre los pesos de los bebés de madres fumadoras o no fumadoras.